### PR TITLE
Support https forward

### LINKED ISSUE #14: Support authentication and authorization for Pulsar Manager (MVP)
Give priority to supporting JWT.

We will use this issue to track authentication and authorization

- [x] Support token management (https://github.com/apache/pulsar-manager/pull/205)
- [x] Support authorization at the namespace level (https://github.com/apache/pulsar-manager/pull/205)
- [x] Support authorization at the topic level (https://github.com/apache/pulsar-manager/pull/205)
- [x] Support for third-party logins such as GitHub (https://github.com/apache/pulsar-manager/pull/222)
- [x] Support management of user (https://github.com/apache/pulsar-manager/pull/226)
- [x] Support role and user binding (https://github.com/apache/pulsar-manager/pull/231)
- [x] Support management of role (https://github.com/apache/pulsar-manager/pull/231)
   - [x] Create, delete, update and list
   - [x] Support the binding of roles and resources
   - [x] Support role isolation between tenants
- [x] Supports cache tenants in the backend (https://github.com/apache/pulsar-manager/pull/231)
- [x] Supports cache namespace in the backend (https://github.com/apache/pulsar-manager/pull/231)
- [ ] Support TLS authentication
- [x] The front end shows the content according to roles (https://github.com/apache/pulsar-manager/pull/238)

### PR DESCRIPTION
**Pull request by tuteng**
Fixed
#14 
#224 
#253

### Motivation
Support https forward.

### Modifications

* Add https forward
* Update readme
* Add option for TLS configurations
* Default enable the user management

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -140,12 +140,20 @@ The pulsar-manager can monitor topics and subscriptions.
 
 3. Access Pulsar manager website.
 
-    Use the account and the password to log in to the Pulsar manager website as below.  
-          
+If you use the container to build the environment, please directly using the the account and the password to log in to the Pulsar manager website as below.
```
Comment (sijie) [inline README.md]: If you are deploying Pulsar Manager 0.1.0 using the released container, you can log in the Pulsar Manager UI using the following credentials.

File: README.md [hunk 2]
```diff
@@ -140,12 +140,20 @@ The pulsar-manager can monitor topics and subscriptions.
...
+
    * Account: `pulsar`  
    * Password: `pulsar`  
+   
+if you use the latest code to deploy the environment yourself, please use the following command to initialize the superuser and password
```
Comment (sijie) [inline README.md]: If you are deploying Pulsar Manager using the latest code, you can create a super-user using the following command.

File: src/README.md [hunk 3]
```diff
@@ -45,6 +45,28 @@ spring.datasource.password=postgres
 ./gradlew -x build -x test
 ```
 
+### Enable Https forward
+
+First, please obtain the client certificate file according to this [document](http://pulsar.apache.org/docs/en/security-tls-transport/). We use the following command to generate the keystore file for Spring Boot.
```
Comment (sijie) [inline src/README.md]: First, please get the client certificate file for connecting to your Pulsar cluster. Please check out this [document](http://pulsar.apache.org/docs/en/security-tls-transport/) on how to create the client certificate file for Pulsar clients. Assume your client certificate is stored in `certs/ca.cert.pem`.

File: src/README.md [hunk 4]
```diff
@@ -45,6 +45,28 @@ spring.datasource.password=postgres
...
+
+```$xslt
+keytool -import -trustcacerts -keystore keystore-file -alias test-keystore -file certs/ca.cert.pem
```
Comment (sijie) [inline src/README.md]:
```suggestion
keytool -import -trustcacerts -keystore keystore-file -alias test-keystore -file [/path/to/client/certs]
```

File: src/README.md (hunk 3 above)
Comment (sijie) [reply src/README.md]: Secondly, following the instructions below to generate a keystore file to be used by Pulsar Manager. Please replace `[/path/to/client/certs]` with the path to your client certificate file.

File: src/README.md [hunk 5]
```diff
@@ -45,6 +45,28 @@ spring.datasource.password=postgres
...
+```
+
```
Comment (sijie) [inline src/README.md]: Thirdly, configure Pulsar Manager to enable TLS by adding the following configuration settings.

**Comment by KannarFr** [comment]
Hi, thanks for the PR :)

Just one question what if the certificates are not locally but "real" certificates, can we just turn TLS on and use the system CAs?

**Comment by KannarFr** [comment]
@tuteng ^

**Comment by tuteng** [comment]
It should be possible. I think you should have such a file `/path/to/ca.cert.pem` after the cluster configuration turns on TLS authentication in order to allow clients to access the cluster. http://pulsar.apache.org/docs/en/security-tls-authentication/#cli-tools  System pulsar-manager also only needs such a file. @KannarFr

**Comment by KannarFr** [comment]
@tuteng Im my case, we are not using cluster as TLS, the cluster nodes are in their own VPN. Nevertheless the proxies (which are the public entry points) are provided using Let's Encrypted generated certificates. So the issue I was having was about query the brokers via the proxies using the public https endpoint from proxies. Basically, it's just a way to make pulsar manager call every routes using https instead of http :p.

I think I missed to precise that points, didn't think about the pulsar TLS auth points, sorry.

**Comment by tuteng** [comment]
@KannarFr Sorry, I mean TLS encryption and pulsar-manager does not support TLS authentication,
http://pulsar.apache.org/docs/en/security-tls-transport/#cli-tools For TLS encryption, pulsar-manager only needs to use the above `ca.cert.pem` file, just like you configured for pulsar admin, In pulsar-manager, you need to convert this file to keystore file through `keytoo` tool. https://github.com/apache/pulsar-manager/blob/master/src/README.md#enable-https-forward

```
keytool -import -trustcacerts -keystore keystore-file -alias test-keystore -file [/path/to/client/certs]
```

File: src/README.md (hunk 3 above)
Comment (gurleen-gks) [reply src/README.md]: @sijie I see support for keystore, 
is there a way to use X509 certificate over keystore?

**Pull request merged**